packages/wallets/particle/src/adapter.ts

```diff
@@ -0,0 +1,185 @@
+
+    private _connecting: boolean;
+    private _wallet: SolanaWallet | null;
+    private _publicKey: PublicKey | null;
+    private _config: ParticleAdapterConfig;
+    private _readyState: WalletReadyState =
+        typeof window === 'undefined' ? WalletReadyState.Unsupported : WalletReadyState.Loadable;
+
+    constructor(config: ParticleAdapterConfig) {
+        super();
+        this._connecting = false;
+        this._publicKey = null;
+        this._wallet = null;
+        this._config = config;
+    }
+
+    get publicKey(): PublicKey | null {
+        return this._publicKey;
+    }
+
+    get connecting(): boolean {
+        return this._connecting;
+    }
+
+    get connected(): boolean {
+        return !!this._publicKey;
+    }
+
+    get readyState(): WalletReadyState {
+        return this._readyState;
+    }
+
+    async connect(): Promise<void> {
+        try {
+            if (this.connected || this.connecting) return;
+            if (this._readyState !== WalletReadyState.Loadable) throw new WalletNotReadyError();
+
+            this._connecting = true;
+
+            let ParticleClass: typeof ParticleNetwork;
+            let WalletClass: typeof SolanaWallet;
+            try {
+                const classes = await Promise.all([
+                    import('@particle-network/auth'),
+                    import('@particle-network/solana-wallet'),
+                ]);
+                ParticleClass = classes[0].ParticleNetwork;
+                WalletClass = classes[1].SolanaWallet;
+            } catch (error: any) {
+                throw new WalletLoadError(error?.message, error);
+            }
+
+            let wallet: SolanaWallet;
+            try {
+                wallet = window.particle || new WalletClass(new ParticleClass(this._config).auth);
```

Comment:
Looking at how this is used, why is it necessary to load two separate SDKs? This will lead to two network calls for files that have to load, and they could be combined because you're just passing one into the other.

Also, what would cause `window.particle` to be set, do you also have a browser extension? And if so, when it's available, can you use it directly without loading an SDK?